### DISCUSSION TITLE
Add PV1 + PV2 and Settings for minimum and maximum charging limit

### DISCUSSION THREAD

**Discussion by bot17** [mtrossbach/noah-mqtt#44] [Ideas]
in the new Server UI we have data for PV1 and PV2 can you add these to the NOAH MQTT?

![image](https://github.com/user-attachments/assets/e4963d4b-ccce-4f93-a490-7d3a56842f92)

**Comment by Dirk-Ahoy** [comment]
Danke - hab ich erst gerade entdeckt das die da einiges nachgerüstet haben - ja das wäre cool -  hab heute nach einigem gebastel  noah-mqtt - telegraf - influx - grafana  ans laufen bekommen, so wird wenigens schon mal was lokal gespeichert am raspberry.  War auch tatsächlich mal etwas Sonne heute das man mal testen konnte :)  Danke für das Projekt.  



![WhatsApp Bild 2024-12-08 um 13 42 16_48f09903](https://github.com/user-attachments/assets/0fadf79c-aed2-402b-806c-f7d3bb2eac10)

**Comment by Crackerliii** [comment]
wäre super, wenn wir diese Entitäten bekommen würden. Im Moment würde man gar nicht mitbekommen, wenn ein Strang keinen Strom mehr liefert.

**Reply by bot17** [reply]
Danke und Grüße von entre

**Comment by msmthng** [comment]
werden die pv daten von den slave akkus auch angezeig? ich habe auch am 1. slave etwas angeschlossen.

**Reply by bot17** [reply]
das kannst du im Webfrontend von Growatt prüfen openapi.growatt.com dort unter Energie

**Reply by msmthng** [reply]
aber da habe ich nur meinen master zur auswahl und der hat pv1 und pv2, die slaves werden dort nicht angezeigt

**Reply by RealGue** [reply]
Aber du schließt doch Module nur am 1. Noah an, insofern passt es doch, dass nur da die Werte angezeigt werden, oder?

**Reply by msmthng** [reply]
Ich habe 4 Module am Noah 1. Noah angeschlossen, aber Du kannst auf jedem Noah Module anschließen, in meinem Fall habe ich aber an den anderen beiden Noahs jeweils 240W Netzteile angeschlossen, so kann ich im Winter alle 2 Wochen dafür sorgen, dass die Noahs, falls sie es wetterbedingt nicht auf 100% schaffen auch mal einen vollen Ladezyklus bekommen und das BMS wieder halbwegs kalibriert.  Und auch das wird mir dann aber bei PV1 und PV2 angezeigt.

Im Falle von Modulen würde man dann ja auch besser sehen wieviel das einzelne Modul bringt.

**Comment by frankfrommelt** [comment]
Well, one can connect modules to all batteries.
And it makes perfect sense.
You have 2 MPP Trackers per battery, so you can connect e.g. 4 panels if
you have two batteries. No need for Y-cables and optimized control of the
panels.


--- Original Nachricht ---
Von: RealGue ***@***.***
Datum: 12. Dezember 2024 23:20:05
An: mtrossbach/noah-mqtt ***@***.***
CC: Subscribed ***@***.***
Betreff: Re: [mtrossbach/noah-mqtt] Add PV1 + PV2 and Settings for minimum
and maximum charging limit (Discussion #44)

**Comment by nschaefer** [comment]
Gibts hier was neues? Hat jemand schon eine Möglichkeit gefunden in Homeassistant den Solar Input aufgeschlüsselt nach PV1 und PV2 abzugreifen?

**Comment by haisch** [comment]
Hallo Zusammen. Das würde auch mir helfen. Insbesondere wenn die Modulspannung in Homeassistant angezeigt werden könnte. So könnte das bekannte 25V Problem vom Noah erkannt werden und bei auftreten der Noah neu gestartet werden. (mittels Switchbot)

**Reply by frankfrommelt** [reply]
Soweit ich weiß werden diese Informationen über die API von Growatt nicht angeboten und können somit von dem AddOn auch nicht zur Verfügung gestellt werden.
   Frank

**Reply by nschaefer** [reply]
Es gibt aber in der REST API voltage und current nach Eingang getrennt. 
Ich zieh mir also einfach die Daten über die REST API und errechne dann mit einem Helfer aus voltage und current die Leistung für pv1 und pv2. Funktioniert wunderbar.
Die Werte könnte man im Addon ja auch verfügbar machen. Die REST API selbst abfragen ist ja auch nicht für jeden so ganz trivial.
